### DISCUSSION TITLE
Web3Modal component in v3

### DISCUSSION THREAD

**Discussion by swkim109** [WalletConnect/walletconnect-monorepo#3779] [Developer Support] [answered]
Can't find `Web3Modal` component in v3. Is it replaced with `createWeb3Modal`?
`createWeb3Modal` does not support several options Web3Modal has, such as `enableNetworkView` and `enableExplorer`.

Any option in v3 for hiding "All Wallets" in the modal?

![wc](https://github.com/WalletConnect/walletconnect-monorepo/assets/46621605/2f28b95f-aa2e-4da7-ae7d-92ec5ea7d14e)

**Comment by glitch-txs** [comment]
1. https://docs.walletconnect.com/web3modal/react/components
2. All Wallets is also used in mobile, removing it would mean removing support for mobile wallets except for those on the main view, do you mean just removing it from the desktop view?

**Reply by swkim109** [reply]
1. I meant [`Web3Modal` component in v2](https://docs.walletconnect.com/web3modal/v2/react/wagmi/components#web3modal). But v3 does not have it, so I guess it is replaced with `createWeb3Modal`. 
2. I just want to show the short list of wallets specified in `featuredWalletIds` for both mobile and desktop. I thought All Wallets just shows  `featuredWalletIds` but it's not. Is it necessary to show all 300+ wallets?

**Reply by glitch-txs** [reply, marked as answer]
You can remove them but if you do the users won't be able to connect with the wallets that you remove on mobile. You can use the same features as in v2
We're currently thinking if this feature should be included or not since it affects the functionality, so some users won't be able to connect to your dapp.
This would be fine on desktop but not on mobile.
What are your thoughts on this?

**Reply by swkim109** [reply]
Well, it depnds on the user's level, crypto-friendly or not. I think users readily install one of the wallets on the list if they don't have it. Of course, some users want to search the wallet already installed. Anyway, I didn't know there are 300+ crypto wallets!
IIRC, `enableExplorer: false` in v2 hides the wallets searching, so I thought v3 also has the option.

**Reply by glitch-txs** [reply]
v3 has the option as well, but it's currently not documented.
